src/OpenTelemetry.Instrumentation.AspNetCore/AspNetCoreInstrumentationOptions.cs

```diff
@@ -34,15 +34,31 @@ public class AspNetCoreInstrumentationOptions
         public Func<HttpContext, bool> Filter { get; set; }
 
         /// <summary>
-        /// Gets or sets an action to enrich an Activity.
+        /// Gets or sets an action to enrich an Activity during start event.
         /// </summary>
         /// <remarks>
         /// <para><see cref="Activity"/>: the activity being enriched.</para>
-        /// <para>string: the name of the event.</para>
-        /// <para>object: the raw object from which additional information can be extracted to enrich the activity.
-        /// The type of this object depends on the event, which is given by the above parameter.</para>
+        /// <para><see cref="HttpRequest"/>: the HttpRequest object from which additional information can be extracted to enrich the activity during start event.</para>
         /// </remarks>
-        public Action<Activity, string, object> Enrich { get; set; }
+        public Action<Activity, HttpRequest> EnrichOnStart { get; set; }
```

Comment:
```suggestion
        public Action<Activity, HttpRequest> EnrichWithHttpRequest { get; set; }
```

Any better? The start/stop, if used, need some explanation to describe what does it refer to

Reply:
EnrichFromHttpRequest

Reply:
Thought the names would provide easier transition from event names as `OnStartActivity`, `OnStopActivity` and `OnException`.

Reply:
How about `EnrichOnRequestStart`, `EnrichOnRequestStop` and `EnrichOnRequestException`?

Reply:
I feel like in .NET "On" is usually at the begging of the event name so maybe `OnRequestStartEnrich`, `OnRequestEndEnrich`, `OnUnhandledExceptionEnrich`? Don't love these but maybe it will inspire better names 😄

Reply:
![image](https://user-images.githubusercontent.com/5232798/196462118-656e8903-423c-4328-b1bd-2cbe86f99f89.png)
The "On" thingie is not really used in many places....

We need the word "Enrich" as the whole point of this callback is to enrich Activity.
WithHttpRequest/FromHttpRequest - to indicate where to get the data to enrich activity.

My votes to the below:
"EnrichFromHttpRequest", "EnrichFromHttpResponse", "EnrichFromException" 
"EnrichWithHttpRequest", "EnrichWithHttpResponse", "EnrichWithException" 

Start/Stop is not really needed as the type would hint about it..Open to other ideas..

Reply:
We decided at the SIG meeting today to go with these names: "EnrichWithHttpRequest", "EnrichWithHttpResponse", "EnrichWithException"